snuba/query/snql/parser.py

```diff
@@ -1537,29 +1543,67 @@ def parse_snql_query(
         with sentry_sdk.start_span(op="processor", description="treeify_conditions"):
             _post_process(query, [_treeify_or_and_conditions], settings)
 
+        # dummy settings and timer dont matter
+        dummy_settings = HTTPQuerySettings()
+        dummy_timer = Timer("snql_pipeline")
+        res = PostProcessAndValidateQuery().execute(
+            QueryPipelineResult(
+                data=(query, dataset, settings, custom_processing),
+                error=None,
+                query_settings=dummy_settings,
+                timer=dummy_timer,
+            )
+        )
+        if res.error:
+            raise res.error
```

Comment:
why raise and then recatch and raise again?

Reply:
Then catch the treeify exception where it's happening and move the exception handling into the pipeline stage

Reply:
im not sure how I would change the structure of the exception catching while maintaining the current behavior